etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/partner/PartnerMetadataOrchestrator.java

```diff
@@ -111,21 +93,21 @@ public void updateMetadataForReceivedMessage(
                 "Updating metadata with sender: {}, timeReceived: {} and hash",
                 sender,
                 timeReceived);
-        PartnerMetadata partnerMetadata =
+        PartnerMetadata updatedPartnerMetadata =
                 new PartnerMetadata(
-                        receivedSubmissionId,
+                        partnerMetadata.receivedSubmissionId(),
                         sender,
                         timeReceived,
                         null,
-                        messageHash,
+                        partnerMetadata.hash(),
                         PartnerMetadataStatus.PENDING,
-                        messageType,
-                        sendingApplicationDetails,
-                        sendingFacilityDetails,
-                        receivingApplicationDetails,
-                        receivingFacilityDetails,
-                        placerOrderNumber);
-        partnerMetadataStorage.saveMetadata(partnerMetadata);
+                        partnerMetadata.messageType(),
+                        partnerMetadata.sendingApplicationDetails(),
+                        partnerMetadata.sendingFacilityDetails(),
+                        partnerMetadata.receivingApplicationDetails(),
+                        partnerMetadata.receivingFacilityDetails(),
+                        partnerMetadata.placerOrderNumber());
```

Comment:
We have an existing paradigm for this using the `.with___()` methods of `PartnerMetadata`.  We should refactor to use those methods or add to them.

That will also make this code more resilient to other changes to `PartnerMetadata`.  If we add additional fields to `PartnerMetadata`, we would need to always come into here and update this constructor.  But what if we forget to update this code?  Data will be accidentally dropped.  So, I think we should pivot to using the `.with___()` methods.